src/load.ts

```diff
@@ -392,6 +395,53 @@ export async function lookUpLatestVersion(
   }
 }
 
+export async function lookUpLatestPatchVersion(
+  packageName: string,
+  version: string,
+  log: LogFunction = () => {}
+): Promise<string> {
+  const customRegistry = getCustomRegistry(log);
+  let res: AxiosResponse;
+  try {
+    if (customRegistry) {
+      res = await axiosGet(`${customRegistry.replace(/\/$/, '')}/${packageName}`, {
+        responseType: 'json'
+      });
+    } else {
+      try {
+        res = await axiosGet(`https://packages.fhir.org/${packageName}`, {
+          responseType: 'json'
+        });
+      } catch (e) {
+        // Fallback to trying packages2.fhir.org
+        res = await axiosGet(`https://packages2.fhir.org/packages/${packageName}`, {
+          responseType: 'json'
+        });
+      }
+    }
+  } catch {
+    throw new LatestVersionUnavailableError(packageName, customRegistry, true);
+  }
+
+  if (res?.data?.versions) {
+    const versions = Object.keys(res.data.versions);
+    const regex = new RegExp(`^${version.split('.').slice(0, -1).join('\\.')}\\.\\d+$`); // regex for matching major.minor version, excluding any qualifiers (like -draft)
+    const matchingVersions = versions.filter(v => v.match(regex));
+    if (matchingVersions.length === 0) {
+      throw new LatestVersionUnavailableError(packageName, customRegistry, true);
+    }
+    const patchVersionNumbers = matchingVersions.map(v =>
+      parseInt(v.slice(v.lastIndexOf('.') + 1))
+    );
+    const latestPatchIndex = patchVersionNumbers.findIndex(
+      p => p === Math.max(...patchVersionNumbers)
+    );
+    return matchingVersions[latestPatchIndex];
```

Comment:
This looks like it should work, but it is a little complicated. What do you think about using [semver](https://www.npmjs.com/package/semver)? I think if we used it, all of above would become:
```js
const latest = maxSatisfying(versions, version, false);
if (latest == null) {
  throw new LatestVersionUnavailableError(packageName, customRegistry, true);
}
return latest;
```

I'm also OK w/ keeping as-is, but if we do that, I think we should check for `/\d+\.\d+\.x/` in this function too since your logic assumes that format.

Reply:
Oh. Wow I double confused myself. I did check that the docs said the default was false, but then when I came back here, I looked at the `true` in the error, so I though "okay so why do I need to explicitly set it to true". Whoops.